rest/handler.go

```diff
@@ -254,3 +281,45 @@ func getLimitFromURL(url *url.URL) (bool, int64, error) {
 
 	return true, limit, nil
 }
+
+type urlHandler struct {
+	caller func(w http.ResponseWriter, r *http.Request, s langserver.HeadlessServer, requestID string)
+	h      *langserverHandler
+}
+
+func newURLHandler(ls *langserverHandler, handler func(w http.ResponseWriter, r *http.Request, s langserver.HeadlessServer, requestID string)) http.Handler {
+	return &urlHandler{h: ls, caller: handler}
+}
+
+func (uh *urlHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
```

Comment:
You could make this a method of `langserverHandler` (with a different name) and pass the caller as an argument.

Reply:
yes

Reply:
see my comment [here](https://github.com/prometheus-community/promql-langserver/pull/120#discussion_r392621983) . sorry for the confusion